### ISSUE TITLE
Mongo nested collection mapping doesn't work for referenced entities

### ISSUE THREAD

**Issue by frynet** [micronaut-projects/micronaut-data#1954]
### Expected Behavior

That's what I want to see:

1. [mongodb document reference](https://www.mongodb.com/docs/manual/reference/database-references/)
2. [similar in spring](https://docs.spring.io/spring-data/mongodb/docs/current-SNAPSHOT/api/org/springframework/data/mongodb/core/mapping/DBRef.html)
3. [raw DBRef (manually filled using mongo db compass)](https://i.stack.imgur.com/9vwAa.png)

### Actual Behaviour

I've tried to declare data entities for my mongo db and all I got is a `null` in a nested (via reference) collection.

[That's what I got (in all 3 cases)](https://i.stack.imgur.com/qjCk8.png)

### Steps To Reproduce

go to my [repository](https://github.com/frynet/MicronautFailedMappingMongoNestedEntities) and follow the instructions

### Environment Information

- OS: Win-10
- JAVA_VERSION: Amazon Coretto 11

### Example Application

https://github.com/frynet/MicronautFailedMappingMongoNestedEntities

### Version

3.7.3

**Comment by radovanradic** [comment]
Try adding this annotation `@Join("bananas")` on `FruitRepoV1`
```
@MongoRepository
@Join("bananas")
interface FruitRepoV1 : CrudRepository<FruitV1, String>
```
 (or findById method) so when `FruitV1` object is loaded from the db it will fetch mapped collection `bananas` (if that is your actual issue here)

**Comment by frynet** [comment]
> Try adding this annotation `@Join("bananas")` on `FruitRepoV1`
> 
> ```
> @MongoRepository
> @Join("bananas")
> interface FruitRepoV1 : CrudRepository<FruitV1, String>
> ```
> 
> (or findById method) so when `FruitV1` object is loaded from the db it will fetch mapped collection `bananas` (if that is your actual issue here)

Still getting `null` :(

**Comment by radovanradic** [comment]
I have created simple test with your model [here](https://github.com/radovanradic/demo-mongo-ns/blob/main/src/test/kotlin/com/example/DemoMongoFruitTest.kt#L22) but not sure if this is your case or you are trying something else. The test is in this sample [repo](https://github.com/radovanradic/demo-mongo-ns) using Micronaut Data MongoDB and Kotlin.

**Comment by dstepanov** [comment]
I think the reporter wants to use `DBRef` mongo type to store the association internally. Right now, we have just a simple object with an id field. We would need to come up with a way to store it differently, maybe it can be stored with `DBRef` when the relationship is annotated with `@BsonRepresentation(BsonType.DB_POINTER)`

**Comment by radovanradic** [comment]
Right, I saw example using relation mapping and was not sure if they wanted to use collection mapping that can be done in Micronaut since DBRef is not supported in Micronaut Data mongoDB. Sorry for adding confusion.
